### ISSUE TITLE
Memory leak on Mixed type dashboard

### ISSUE THREAD

**Issue by korservick** [grafana/grafana#43998]
**What happened**:
Dashboard with "-- Mixed --" datasource type and no query row in v8.3.3 after apply (or save and load) cause memory leak brouser.

**What you expected to happen**:
Just empty dashboard with no data

**How to reproduce it (as minimally and precisely as possible)**:
1. Create Dashboard
2. Add new panel
3. Change data source to "-- Mixed --"
4. Remove all query (just "A" for new panel)
5. Apply

**Anything else we need to know?**:
Saved dashboard:

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations \u0026 Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 1912,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "datasource",
        "uid": "-- Mixed --"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "palette-classic"
          },
          "custom": {
            "axisLabel": "",
            "axisPlacement": "auto",
            "barAlignment": 0,
            "drawStyle": "line",
            "fillOpacity": 0,
            "gradientMode": "none",
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineInterpolation": "linear",
            "lineWidth": 1,
            "pointSize": 5,
            "scaleDistribution": {
              "type": "linear"
            },
            "showPoints": "auto",
            "spanNulls": false,
            "stacking": {
              "group": "A",
              "mode": "none"
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 8,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom"
        },
        "tooltip": {
          "mode": "single"
        }
      },
      "targets": [],
      "title": "Panel Title",
      "type": "timeseries"
    }
  ],
  "refresh": "",
  "schemaVersion": 34,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "leak",
  "uid": "XMdQK517z",
  "version": 3,
  "weekStart": ""
}
```

**Environment**:
- Grafana version: v8.3.3
- Data source type & version: "-- Mixed --"
- OS Grafana is installed on: helm k3s
- User OS & Browser: Safari, Chrome
- Grafana plugins:
- Others:

**Comment by korservick** [comment]
Grafana v7.5.11 does not have this bug

**Comment by mckn** [comment]
I have verified the behavior but I don't know what the root cause is.

**Comment by fatoh2** [comment]
@kaydelaney I see this one was merged already, was it included in the 8.3.4 release?
